src/main/java/org/apache/commons/csv/CSVParser.java

```diff
@@ -445,14 +446,25 @@ public void close() throws IOException {
                 new LinkedHashMap<>();
     }
 
+    private static final class Headers {
+        final Map<String, Integer> headerMap;
+        final List<String> headerNames;
+
+        Headers(Map<String, Integer> headerMap, List<String> headerNames) {
+            this.headerMap = headerMap;
+            this.headerNames = headerNames;
+        }
+    }
+    
     /**
      * Creates the name to index mapping if the format defines a header.
      *
      * @return null if the format has no header.
      * @throws IOException if there is a problem reading the header or skipping the first record
      */
-    private Map<String, Integer> createHeaderMap() throws IOException {
+    private Headers createHeaderMapAndHeaderNames() throws IOException {
```

Comment:
I would rename createHeaderMapAndHeaderNames() to createHeaders() to match the class name.